packages/aws-cdk/lib/cli/cli.ts

```diff
@@ -546,6 +547,28 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
   }
 }
 
+export async function displayFlagsMessage(ioHost: CliIoHost, toolkitStackName: string, cloudExecutable: CloudExecutable,
```

Comment:
You can move this logic inside `CdkToolkit`, where you already have access to the `Toolkit` instance.